kaldb/src/main/java/com/slack/kaldb/logstore/LuceneIndexStoreImpl.java

```diff
@@ -143,15 +145,37 @@ public void run() {
         "Created a lucene index {} at: {}", id, indexDirectory.getDirectory().toAbsolutePath());
   }
 
+  /**
+   * Attempts to determine an optimal ram buffer size based on the size of the heap. The target of
+   * 10% matches that of the defaults of ES.
+   *
+   * @see {https://www.elastic.co/guide/en/elasticsearch/reference/current/indexing-buffer.html}
+   */
+  protected static long getRAMBufferSizeMB(long heapMaxBytes) {
+    long targetBufferSize = 256;
+    if (heapMaxBytes != Long.MAX_VALUE) {
+      targetBufferSize = Math.min(2048, Math.round(heapMaxBytes / 1e6 * 0.10));
+    }
+    LOG.warn(
+        "Setting max ram buffer size to {}mb, heap max bytes detected as {}",
+        targetBufferSize,
+        heapMaxBytes);
+    return targetBufferSize;
+  }
+
   private IndexWriterConfig buildIndexWriterConfig(
       Analyzer analyzer,
       SnapshotDeletionPolicy snapshotDeletionPolicy,
       LuceneIndexStoreConfig config,
       MeterRegistry metricsRegistry) {
+    long ramBufferSizeMb = getRAMBufferSizeMB(Runtime.getRuntime().maxMemory());
+    boolean useCFSFiles = ramBufferSizeMb <= 128;
```

Comment:
Let's do a static final for the ram buffer size along with a comment why 128 is a good value to use.